### PR TITLE
[12.0][MIG] l10n_us_gaap

### PR DESCRIPTION
**Pull request by dduarte-odoogap**

### CODE REVIEW

**Comment by max3903** [comment]
@dduarte-odoogap Can you split the readme file in a readme folder:
https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

and add the `development_status` and `maintainers` keys in the `__manifest__`:
https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

**Comment by dduarte-odoogap** [comment]
@max3903 i will do that soon

**Comment by bodedra** [comment]
@dduarte-odoogap opened a PR https://github.com/odoogap/l10n-usa/pull/1 to attend @max3903's comment. Could you review?

File: l10n_us_gaap/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    'name': 'United States Sample GAAP Chart of Accounts',
+    'version': '12.0.1.0.0',
+    'development_status': 'Mature',
```
Comment (max3903) [inline l10n_us_gaap/__manifest__.py:4]: @dduarte-odoogap We need to add unit tests to reach `Stable`:
https://odoo-community.org/page/module-maturity-levels

**Comment by dduarte-odoogap** [comment]
@max3903 just added the first unit tests

File: l10n_us_gaap/tests/test_base_us.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class TestChartAcount(TransactionCase):
+
+    def test_basic(self):
+        chart_us = self.env.ref('l10n_us_gaap.l10n_us_gaap_chart_template')
+        chart_us.try_loading_for_current_company()
+        bank_account = self.env['account.account'].search([('code', '=', '111101')])
```
Comment (bodedra) [inline l10n_us_gaap/tests/test_base_us.py]:
```suggestion
        bank_account = self.env['account.account'].search(
            [('code', '=', '111101')])
```

File: l10n_us_gaap/tests/test_base_us.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        liquidity_transfer = self.env['account.account'].search([('code', '=', '111901')])
```
Comment (bodedra) [inline l10n_us_gaap/tests/test_base_us.py]:
```suggestion
        liquidity_transfer = self.env['account.account'].search(
            [('code', '=', '111901')])
```

File: l10n_us_gaap/tests/test_base_us.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        finished_goods = self.env['account.account'].search([('code', '=', '133000')])
```
Comment (bodedra) [inline l10n_us_gaap/tests/test_base_us.py]:
```suggestion
        finished_goods = self.env['account.account'].search(
            [('code', '=', '133000')])
```

File: l10n_us_gaap/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,17 @@
+This module provides the capability to execute a cycle count strategy in a
```
Comment (bodedra) [inline l10n_us_gaap/readme/DESCRIPTION.rst]: Please remove all content from these file and keep empty Description.rst file.

File: l10n_us_gaap/tests/test_base_us.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        self.assertEqual(bank_account.name, 'Bank')
```
Comment (bodedra) [inline l10n_us_gaap/tests/test_base_us.py:46]: These line is failed. Can you check these.

Screenshot_2019-07-15 Job #219 3 - OCA l10n-usa - Travis CI

**Review by bodedra — CHANGES_REQUESTED** [review]

**Comment by dduarte-odoogap** [comment]
@bodedra i run tests locally with success using :

`-d v12_us_test --test-enable -i l10n_us_gaap`

can you kindly point me to the script that runs OCA tests?

**Comment by dduarte-odoogap** [comment]
@bodedra and @max3903 found the travis script and seems there is a problem, when the CI starts it creates a DB with the following modules:

uom,product,bus,portal,base_setup,web,web_tour,contacts,analytic,decimal_precision,mail,http_routing,base,account,digest

These have as a dependency:
l10n_generic_coa

Now, if i try on my local i can successfully run the tests directly with :
`./odoo-bin -d v12_us_test --test-enable -i l10n_us_gaap`

Tests are successful as the COA is installed by default, but trying to replace the exiting CoA seems to be hard as all the default CoA accounts are already created.

**Comment by bodedra** [comment]
@max3903 @pedrobaeza @jbeficent Would you please help us here.

FYI, but not sure is these relevant or not.

https://github.com/OCA/maintainer-quality-tools/issues/412

**Comment by pedrobaeza** [comment]
@bodedra I always make tests over a new company populated in them, without having any problem. Check for example https://github.com/OCA/l10n-spain/blob/5ccd40ba64bbb57b7ef36b29986944e3436220ab/l10n_es_aeat/tests/test_l10n_es_aeat_mod_base.py#L38

**Comment by bodedra** [comment]
Thanks @pedrobaeza 

@dduarte-odoogap I have fixed test cases at https://github.com/odoogap/l10n-usa/pull/4 Would you please merge it.

**Review by bodedra — APPROVED** [review]
LGTM :+1: 

@dduarte-odoogap Can you squash your commits into one and my contribution into another one ? So at the end, we have two commits for these migration.

**Comment by bodedra** [comment]
@max3903 @jbeficent Can you provide your feedback on these?

**Review by JordiBForgeFlow — APPROVED** [review]
Tested. Works as expected.

**Comment by JordiBForgeFlow** [comment]
This is good to merge for me.

**Pull request closed without merging**